src/mscorlib/mscorlib.csproj

```diff
@@ -181,7 +181,8 @@
   </PropertyGroup>
 
   <Import Project="$(MSBuildThisFileDirectory)Tools\Versioning\GenerateVersionInfo.targets"/>
-  <Import Project="GenerateSplitStringResources.targets"/>
+  <Import Project="GenerateSplitStringResources.Windows.targets" Condition="'$(OsEnvironment)'=='Windows_NT'"/>
```

Comment:
I am fine with this staged approach provided you will unify the files so that we have a single file to maintain.